rasterio/warp.py

```diff
@@ -169,6 +176,7 @@ def transform_bounds(
 
 
 @ensure_env
+@require_gdal_version('2.0', param='resampling', values=GDAL2_RESAMPLING)
```

Comment:
I like this a lot, @brendan-ward.

It is based on a model of GDAL where the number of features increases as the version increases. We wouldn't be able to require a maximum version of GDAL for a feature with the same decorator, right? 

GDAL rarely removes any features, so this isn't a deal breaker for me 😄

Reply:
@sgillies I think we could add an additional parameter like `is_max_version` to indicate this case.

It seems like `antimeridian_cutting` in `warp.py::transform_geom` would fit the case for this pattern, since that parameter is always `True` as of GDAL 2.2.0.  However, if we catch this in the decorator, I'd probably want to return a `GDALVersionError` instead of `GDALBehaviorChangeException` that we return now for this case.

```
@requires_gdal_version('2.1', param=antimeridian_cutting, 
                       values=[False], is_max_version=True)
def transform_geom(...)
```
would evaluate as `2.1` is the last version where `False` is allowed for `antimeridian_cutting`.